docs/DeveloperGuide.md

```diff
@@ -1,38 +1,584 @@
+
+### Family Component
+
+The **Class** of this component is specified in [`Family.java`](
+https://github.com/AY2122S2-CS2113T-T10-2/tp/blob/master/src/main/java/seedu/planitarium/person/Family.java)
+, [`PersonList.java`](
+https://github.com/AY2122S2-CS2113T-T10-2/tp/blob/master/src/main/java/seedu/planitarium/person/PersonList.java)
+and [`Person.java`](
+https://github.com/AY2122S2-CS2113T-T10-2/tp/blob/master/src/main/java/seedu/planitarium/person/Person.java)
+
+![FamilyComponent](images/FamilyComponent.png)
+
+The list of persons consists of a `Family` that is made up of generational `PersonList`s. Each `PersonList` holds a 
+list of `Person`s who belong to that generation.
+
+The `Person` component,
+
+* Stores the logical grouping of persons added i.e., all `Person` objects must belong to one of the `PersonList`s.
+* Stores the total number of `Person`s in the `Family`, as well as each of the generational `PersonList`s.
+* Depends on the `Money` component to help keep track of each `Person`'s income and expenditure as each `Person` 
+  contains an `IncomeList` and `ExpenditureList`.
+
+### Money Component
+
+<!-- {For Jiarong} -->
+
+### Storage Component
+
+**Class:** [`Storage.java`
+](https://github.com/AY2122S2-CS2113T-T10-2/tp/blob/master/src/main/java/seedu/planitarium/storage/Storage.java)
+
+![StorageDiagram](images/StorageDiagram.png)
+
+The `Storage` component,
+
+* can save each person's data and their income and expenditure data when the program exits into a local file, and reads
+  them back into corresponding objects when the program runs again.
+* depends on the classes in the `Person` component (because the `Storage` component's job is to save/retrieve objects to
+  the `Person`)
+
+---
+
+## Implementation
+
+This section describes some noteworthy details on how certain features are implemented.
+
+### Command Execution
+
+In PlanITarium, a command usually has the following format:
+
+```md
+[command type][indicator][description]
+```
+
+For example,
+
+```md
+add \n Alice
+```
+
+is to add a person whose name is Alice. `add` here is the command type to add a person, and `\n` is the indicator of the
+name of person.
+
+There may be several indicators for every input, such as
+
+```md
+addin \i 200 \d salary \u 1
+```
+
+is to add an income for the person whose uid is 1, and the description of this income is salary.
+
+When `PlanITarium` receives such input, it will pass the input to `CommandFactory`. The `CommandFactory` will call
+`Parser` to parse the input into several components according to the indicator. The `Parser` will then return the type
+of command to `CommandFactory`. According to the type, `CommandFactory` will return a command object to `PlanITarium`.
+After receiving the command, `PlanITarium` will call `execute()` of command object to execute the command.
+
+The following diagram is the sequence diagram of this entire process.
+
+![CommandFactorySequence](images/CommandFactorySequence.png)
```

Comment:
perhaps the return type of Parser can be included in the diagram ? currently it's just return with dotted line, I think it is fine based on google, but you may want to write "Command" instead so that it is clear that a command object is returned...Can look into it!